rcl/src/rcl/logging.c

```diff
@@ -64,10 +69,17 @@ rcl_logging_ext_lib_output_handler(
   const char * format, va_list * args);
 
 rcl_ret_t
-rcl_logging_configure(const rcl_arguments_t * global_args, const rcl_allocator_t * allocator)
+rcl_logging_init(const rcl_arguments_t * global_args, const rcl_allocator_t * allocator)
 {
+  pthread_mutex_lock(&g_rcl_logging_mutex);
```

Comment:
Do we really need a mutex?
Maybe, using an atomic?
We have multiplatform support for atomics https://github.com/ros2/rcutils/blob/cd971e01c18d9646c80d73dec7da94af2a8e489b/include/rcutils/stdatomic_helper.h#L88.

Reply:
yeah, okay using atomic counter provided rcutils.

and supporting the following case as well.

> we should ensure that the logger is not finished until the last context is finished. So we should probably do some kind of "reference" counting, instead of just a boolean indicating if it was already init or not.